src/osd/osd_types.cc

```diff
@@ -2242,12 +2251,22 @@ void pg_pool_t::decode(ceph::buffer::list::const_iterator& bl)
     last_force_op_resend = last_force_op_resend_prenautilus;
     pg_autoscale_mode = pg_autoscale_mode_t::WARN;    // default to warn on upgrade
   }
-  if (struct_v >= 30) {
+  if (struct_v == 30) {
     decode(peering_crush_bucket_count, bl);
     decode(peering_crush_bucket_target, bl);
     decode(peering_crush_bucket_barrier, bl);
     decode(peering_crush_mandatory_member, bl);
   }
+  if (struct_v >= 31) {
+    bool maybe_peering_crush_data;
```

Comment:
```suggestion
  std::optional<the_new_struct_here> stretch_related_stuff;
  decode(stretch_related_stuff, bl);
```

Reply:
Since i didn't created inner struct, i still need to do some process on the returned optional. adding it